pinax/stripe/models.py

```diff
@@ -253,7 +292,12 @@ def status(self):
 
     @property
     def stripe_invoice(self):
-        return stripe.Invoice.retrieve(self.stripe_id)
+        return stripe.Invoice.retrieve(
+            self.stripe_id,
+            stripe_account=(
+                self.customer.stripe_account if self.customer_id else None
```

Comment:
Better to use `hasattr(self, 'customer')` here / or a `try/except self._meta.model.customer.RelatedObjectDoesNotExist`?!

Reply:
Care to explain why those more verbose approaches would be better? :)